python/ray/data/_internal/logical/optimizers.py

```diff
@@ -19,30 +30,211 @@ class PhysicalOptimizer(Optimizer):
 
     @property
     def rules(self) -> List["Rule"]:
-        # TODO: Add physical optimizer rules.
-        return []
+        return [OperatorFusionRule()]
 
 
-class LogicalPlan:
-    """The plan with a DAG of logical operators."""
-
-    def __init__(self, dag: LogicalOperator):
-        self._dag = dag
-
-    @property
-    def dag(self) -> LogicalOperator:
-        """Get the DAG of logical operators."""
-        return self._dag
-
-
-def get_execution_dag(logical_dag: LogicalOperator) -> PhysicalOperator:
-    """Get the DAG of physical operators to execute.
+def get_execution_plan(logical_plan: LogicalPlan) -> PhysicalPlan:
+    """Get the physical execution plan for the provided logical plan.
 
     This process has 3 steps:
     (1) logical optimization: optimize logical operators.
     (2) planning: convert logical to physical operators.
     (3) physical optimization: optimize physical operators.
     """
-    optimized_logical_dag = LogicalOptimizer().optimize(logical_dag)
-    physical_dag = Planner().plan(optimized_logical_dag)
-    return PhysicalOptimizer().optimize(physical_dag)
+    logical_plan = LogicalOptimizer().optimize(logical_plan)
+    physical_plan = Planner().plan(logical_plan)
+    return PhysicalOptimizer().optimize(physical_plan)
+
+
+class OperatorFusionRule(Rule):
```

Comment:
Can we move this rule into a separate file? That would make code easier to read.

Reply:
Sure! That was the eventual plan once we had more than one rule, but I can do that now!